src/Nethermind/Nethermind.Db/Blooms/FixedSizeFileStore.cs

```diff
@@ -58,47 +51,19 @@ public void Write(long index, ReadOnlySpan<byte> element)
 
         public int Read(long index, Span<byte> element)
         {
-            EnsureFlushed();
-
-            lock (_fileRead)
-            {
-                SeekIndex(_fileRead, index);
-                return _fileRead.Read(element);
-            }
+            return RandomAccess.Read(_file, element, GetPosition(index));
```

Comment:
I think this may result in a torn read if the element is quite bit and is being written at the moment. I don't know semantics of the usage but not sure if the reading without a lock is right here. Could you elaborate @benaadams ?

Reply:
Wasn't using a shared lock with the writer, only between reads so no sematic change.

Previously it needed a lock as it did a seek then read from seeked location
```
SeekIndex(_fileRead, index);
return _fileRead.Read(element);
```
Now it does "read at location" and doesn't need to maintain that seeked state for the read